### PR TITLE
Change api, Add support for s3 bucket resource and better cft loader

### PR DESCRIPTION
**Pull request by sigmabaryon**

### CODE REVIEW

File: pkg/iac-providers/cft/v1/load-dir_test.go [hunk 1]
```diff
@@ -107,24 +113,50 @@ func TestLoadIacDir(t *testing.T) {
...
+						}
+					}
+					if !found {
+						t.Errorf("resource %s of type %s from test data not found", testResource.ID, name)
+					}
+				}
+			}
+
+			mGot, ok := gotErr.(*multierror.Error)
+			fmt.Printf("gotErr: %+v", mGot)
```
Comment (bagulm123) [inline pkg/iac-providers/cft/v1/load-dir_test.go]: you may want to remove this logger

File: pkg/iac-providers/cft/v1/testdata/deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+Resources:
```
Comment (bagulm123) [inline pkg/iac-providers/cft/v1/testdata/deploy.yaml:1]: Why the content of S3 resource is here? It should be part of KaiMonkey, isn't it? or I am missing on something here?

**Review by bagulm123 — COMMENTED** [review]
2 comments except that everything LGTM

File: pkg/iac-providers/cft/v1/testdata/deploy.yaml (hunk 2 above)
Comment (sigmabaryon) [reply pkg/iac-providers/cft/v1/testdata/deploy.yaml:1]: Its for load-file_test and not load-dir

File: pkg/iac-providers/cft/v1/load-dir_test.go (hunk 1 above)
Comment (sigmabaryon) [reply pkg/iac-providers/cft/v1/load-dir_test.go]: done

**Pull request closed without merging**